src/Mgate/TresoBundle/Form/Type/BVType.php

```diff
@@ -35,7 +35,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ->add('mission', Select2EntityType::class, array(
                       'label' => 'Mission',
                        'class' => 'Mgate\\SuiviBundle\\Entity\\Mission',
-                       'choice_label' => 'nom',
+                       'choice_label' => 'reference',
```

Comment:
That commit basically remove changes done in 41283abb86da49c18cfe50d528f5cad0d1c51648

A compromise is to inject the namingConvention parameter in that form  (through an option, it will be ok) and use it as value of that form.  It has to be a robust solutions, thus add reference as default value in the OptionResolver